scripts/commands/build.js

```diff
@@ -86,17 +81,24 @@ async function buildRelease( currentBranchName ) {
 	}
 
 	if ( ! buildSuccess ) {
-		error( `npm run build failed (${ statusLine })` );
-		error( stderr );
-		return false;
+		return abortAndSwitchToBranch(
+			`npm run build failed (${ statusLine })`,
+			'error',
+			currentBranchName
+		);
 	}
 
 	// Verify that the build is valid.
 	const verificationErrors = await verifyBuild();
 	if ( verificationErrors.length > 0 ) {
-		error( 'Build verification failed:' );
-		verificationErrors.map( ( err ) => error( `\t${ err }` ) );
-		return false;
+		let errorDetails = '';
+		verificationErrors.map( ( err ) => ( errorDetails += `\t${ err }` ) );
+
+		return abortAndSwitchToBranch(
+			`Build verification failed:\n${ errorDetails }`,
+			'error',
+			currentBranchName
+		);
 	}
 
 	// Delete the node_modules directory so they don't get packaged up.
```

Comment:
It would be helpful if we tagged on a message to note that users need to (re)run `npm install` after we've deleted `node_modules`.